packages/core/src/components/date-picker/date-picker.tsx

```diff
@@ -781,8 +782,18 @@ export class DatePicker {
                       id={`day-cell-${day}`}
                       date-calender-day
                       class={this.getDayClasses(day)}
-                      onClick={() => this.selectDay(day)}
-                      onKeyUp={(e) => e.key === 'Enter' && this.selectDay(day)}
+                      onClick={(e) => {
+                        const target = e.currentTarget as HTMLElement;
+                        if (!target.classList.contains('disabled')) {
+                          this.selectDay(day);
+                        }
+                      }}
+                      onKeyUp={(e) => {
+                        const target = e.currentTarget as HTMLElement;
+                        if (!target.classList.contains('disabled')) {
+                          e.key === 'Enter' && this.selectDay(day);
+                        }
+                      }}
```

Comment:
`selectDay` is only used from the event listener in this case you can pass `target` element to the selectDay and check the disable condition there. This prevents duplicated code